core/components/ui/counter/counter.tsx

```diff
@@ -64,6 +54,10 @@ export const Counter = forwardRef<ElementRef<'div'>, CounterProps>(
   ) => {
     const [value, setValue] = useState<number | ''>(getDefaultValue(defaultValue, min, max));
     const inputRef = useRef<ElementRef<'input'>>(null);
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions, @typescript-eslint/non-nullable-type-assertion-style
+    useImperativeHandle(ref, () => inputRef.current as ElementRef<'input'>);
```

Comment:
Is there a better way to do this? I'm trying to use ref in sibling elements.

Reply:
Using this is fine tbh, but you can type it better like so:
```ts
type CounterRef = ElementRef<'input'> | null;

const inputRef = useRef<CounterRef>(null);

useImperativeHandle<CounterRef, CounterRef>(ref, () => inputRef.current);
```

Reply:
This seems to work perfectly, thanks @chanceaclark